packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/__tests__/ExpoSQLiteDatabase.test.ts

```diff
@@ -0,0 +1,416 @@
+			mockTransaction.executeSql.mockImplementation(
+				(statement, params, successCallback, errorCallback) => {
+					errorCallback(mockTransaction, MOCK_SQL_ERROR);
+				}
+			);
+
+			await expect(
+				instance.get(DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS)
+			).rejects.toEqual(MOCK_SQL_ERROR);
+		});
+	});
+
+	describe('save', () => {
+		it('should resolve promise and return null when transaction is successfull', async () => {
+			mockTransaction.executeSql.mockImplementation(
+				(statement, params, successCallback, errorCallback) => {
+					successCallback(mockTransaction, MOCK_RESULT_SET_NUMBER);
+				}
+			);
+			await expect(
+				instance.save(DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS)
+			).resolves.toBeNull();
+		});
+
+		it('should reject promise and return error when the transaction is unsuccessfull', async () => {
+			mockTransaction.executeSql.mockImplementation(
+				(statement, params, successCallback, errorCallback) => {
+					errorCallback(mockTransaction, MOCK_SQL_ERROR);
+				}
+			);
+
+			await expect(
+				instance.save(DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS)
+			).rejects.toEqual(MOCK_SQL_ERROR);
+		});
+	});
+
+	describe('batchQuery', () => {
+		const queryParameterizedStatements: Set<ParameterizedStatement> = new Set([
+			[DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS],
+			[DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS],
+		]);
+
+		it('should resolve promise and return first element of each query result', async () => {
+			mockTransaction.executeSql
+				.mockImplementationOnce(
+					(statement, params, successCallback, errorCallback) => {
+						successCallback(mockTransaction, MOCK_RESULT_SET_NUMBER);
+					}
+				)
+				.mockImplementation(
+					(statement, params, successCallback, errorCallback) => {
+						successCallback(mockTransaction, MOCK_RESULT_SET_STRING);
+					}
+				);
+
+			await expect(
+				instance.batchQuery(queryParameterizedStatements)
+			).resolves.toEqual([
+				MOCK_RESULT_ARRAY_NUMBER[0],
+				MOCK_RESULT_ARRAY_STRING[0],
+			]);
+		});
+
+		it('should reject promise and return error when the transaction is unsuccessfull', async () => {
+			mockTransaction.executeSql.mockImplementation(
+				(statement, params, successCallback, errorCallback) => {
+					errorCallback(mockTransaction, MOCK_SQL_ERROR);
+				}
+			);
+
+			await expect(
+				instance.batchQuery(queryParameterizedStatements)
+			).rejects.toEqual(MOCK_SQL_ERROR);
+		});
+	});
+
+	describe('batchSave', () => {
+		const saveParameterizedStatements: Set<ParameterizedStatement> = new Set([
+			[DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS],
+			[DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS],
+		]);
+
+		const deleteParameterizedStatements: Set<ParameterizedStatement> = new Set([
+			[DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS],
+			[DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS],
+		]);
+
+		it('multiple save statements, no delete statements. should resolve to null when successfull', async () => {
```

Comment:
Nit: Rephrase these so they read more like a sentence
```
it('should resolve to null when successful when provided multiple save statements', async () => {
```
Or you can try and group these cases under describe blocks
```js
describe('multiple save statements, no delete statements', () => {
  it('should resolve ...', () => {})
  it('should reject ...', () => {})
})
```